temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
+        if isinstance(features, np.ndarray):
+            features = [features]
+
+        self.features = features
+        self.timestamps = timestamps
+
+    def __len__(self) -> int:
+        return len(self.timestamps)
+
 
 class NumpyEvent:
+    # tolerance levels for comparing float features in __eq__
+    rtol: float = 1e-9
+    atol: float = 1e-9
+
     def __init__(
         self,
-        data: Dict[Tuple, List[NumpyFeature]],
-        sampling: NumpySampling,
+        data: Dict[Tuple, IndexData],
+        feature_names: Union[str, List[str]],
+        index_names: Union[str, List[str]],
+        is_unix_timestamp: bool = False,
     ) -> None:
         self._data = data
-        self._sampling = sampling
+        self._feature_names = (
+            [feature_names] if isinstance(feature_names, str) else feature_names
+        )
+        self._index_names = (
+            [index_names] if isinstance(index_names, str) else index_names
+        )
+        self._is_unix_timestamp = is_unix_timestamp
 
     @property
-    def data(self) -> Dict[Tuple, List[NumpyFeature]]:
+    def data(self) -> Dict[Tuple, IndexData]:
         return self._data
 
     @property
-    def sampling(self) -> NumpySampling:
-        return self._sampling
+    def feature_names(self) -> List[str]:
+        return self._feature_names
 
     @property
-    def _first_index_features(self) -> List[NumpyFeature]:
-        if self.first_index_value() is None:
-            return []
-        return self.data[self.first_index_value()]
+    def index_names(self) -> List[str]:
+        return self._index_names
+
+    @property
+    def is_unix_timestamp(self) -> bool:
+        return self._is_unix_timestamp
+
+    @property
+    def _first_index_features(self) -> List[np.ndarray]:
+        return list(self._data.values())[0].features
 
     @property
     def dtypes(self) -> Dict[str, DType]:
         return {
-            feature.name: feature.dtype
-            for feature in self._first_index_features
+            feature_name: DTYPE_MAPPING[feature.dtype.type]
+            for feature_name, feature in zip(
+                self._feature_names, self._first_index_features
+            )
         }
 
-    @sampling.setter
-    def sampling(self, sampling: NumpySampling) -> None:
-        self._sampling = sampling
+    @property
+    def feature_count(self) -> int:
+        return len(self._feature_names)
 
-    def first_index_value(self) -> Optional[Tuple]:
-        if self.data is None or len(self.data) == 0:
-            return None
+    def copy(self) -> NumpyEvent:
+        return deepcopy(self)
 
-        return next(iter(self.data))
+    def iterindex(self) -> Iterable[Tuple[Tuple, IndexData]]:
+        yield from self.data.items()
 
-    def first_index_features(self) -> List[NumpyFeature]:
-        if self.first_index_value() is None:
-            return []
-        return self.data[self.first_index_value()]
+    def index_dtypes(self) -> Dict[str, DType]:
```

Comment:
Can we have "dict" somewhere in the function name?

Do we really need all those accessors (iterator, extract dtype to dict, extract name to list, etc.). I think the code (both of this method and of the callers) would improve by having a single simple accessor.

Reply:
I agree, it's a placeholder function (we need to improve index handling in the numpy backend in a future PR). I'll add a TODO as a reminder